### ISSUE TITLE
Questions and concerns about DINOv2 fine-tuning

### ISSUE THREAD

**Issue by smandava98** [facebookresearch/dinov2#245]
Hi,

Thanks for the great work. I am trying to extract visual features from various frames in a video for video analysis. I tried inputting some of the frames into the web demo to see if it does downstream tasks like depth/semantic segmentation well and it does okay but not great. 

In your paper, you mentioned DINOv2 does not need finetuning but I am starting to believe at least a small bit of finetune may be necessary for my tasks. I have two main questions:

- how can I efficiently fine-tune DINOv2? I know loading pre-trained weights and my own data and just running the training process is a possible step but I am just not sure how much this will affect the model. I have on the scale of ~300k images that can benefit from self supervision.

- how can I incorporate temporal info? Is weight inflation the recommended approach or is there an alternative?

Any help would be greatly appreciated and thanks for the wonderful work.

My goal is to build a bespoke video classification model that can take in video frames and just output bounding boxes. I am just confused on how to use DINO well in this case because I do not have a clear understanding of how vision foundation models work.

**Comment by qasfb** [comment]
So if you want to do fine-tuning, you just need to build a setup that uses a ViT backbone, then initialize it with the DINOv2 weights; then, you need to train it along with the rest of your model on your task, and in order to keep the benefits of the pretraining, I recommend using small learning rates on the backbone. You can maybe take a look at Deit-3 (https://arxiv.org/abs/2204.07118) and inspire from how they do it.

It can be also easier to fine-tune only a few (let's say, 4) last layers of the pretrained ViT. I believe this is what they do in MAE (masked autoencoders).

In general, fine-tuning is not really part of our expertise, so I don't think we can help much on this.

**Comment by smandava98** [comment]
Awesome. Thank you for the response!

**Comment by amundra15** [comment]
> It can be also easier to fine-tune only a few (let's say, 4) last layers of the pretrained ViT.
I am trying to implement this, but am running into errors. I have two approaches in mind:
1. Set the requires_grad flag to false for the initial layers. However, upon passing this model to FSDP, it complains that the requires_grad for all layers need to be uniform.
2. Set the LR for the initial layers to 0. However, I am not able to figure out how various layers are mapped to the params_group using get_params_groups().

Any help would be appreciated! @qasfb
